.github/workflows/test.yml

```
@@ -26,81 +26,57 @@ jobs:
       - name: Setup Environment
         uses: ./.github/actions/setup
 
-      - name: Install Compact compiler
-        id: setup
+      - name: Install Compact developer tools
         shell: bash
         run: |
-          set -euo pipefail
-          # Create directory for compiler
-          COMPACT_HOME="$HOME/compactc"
-          mkdir -p "$COMPACT_HOME"
-
-          # Create URL
-          ZIP_FILE="compactc_v${COMPILER_VERSION}_x86_64-unknown-linux-musl.zip"
-          DOWNLOAD_URL="https://d3fazakqrumx6p.cloudfront.net/artifacts/compiler/compactc_${COMPILER_VERSION}/${ZIP_FILE}"
-
-          echo "⬇️ Downloading Compact compiler..."
-          curl -Ls "$DOWNLOAD_URL" -o "$COMPACT_HOME/compactc.zip"
-
-          echo "📦 Extracting..."
-          unzip -q "$COMPACT_HOME/compactc.zip" -d "$COMPACT_HOME"
-          chmod +x "$COMPACT_HOME"/{compactc,compactc.bin,zkir}
-
-          echo "📁 Setting environment variables..."
-          echo "COMPACT_HOME=$COMPACT_HOME" >> "$GITHUB_ENV"
-          echo "$COMPACT_HOME" >> "$GITHUB_PATH"
-
-          echo "✅ Verifying installation..."
-          if [ ! -f "$COMPACT_HOME/compactc" ]; then
-            echo "::error::❌ compactc not found in $COMPACT_HOME"
-            exit 1
-          fi
+          curl --proto '=https' --tlsv1.2 -LsSf https://github.com/midnightntwrk/compact/releases/latest/download/compact-installer.sh | sh
+          echo "$HOME/.compact/bin" >> "$GITHUB_PATH"
 
+      - name: Install and verify toolchain
+        run: |
+          compact update ${{ env.COMPILER_VERSION }}
           echo "🤖 Testing installation..."
-          "$COMPACT_HOME/compactc" --version
+          compact compile --version
+          compact compile --language-version
 
       - name: Check compiler and language version
         run: |
-            COMPILER_OUTPUT=$(compactc --version)
-            COMPUTED_COMPILER_VERSION=$(echo "$COMPILER_OUTPUT" | grep -oP '\b0\.[0-9]+\.[0-9]+\b' | head -n 1)
-            if [ "$COMPUTED_COMPILER_VERSION" != "$COMPILER_VERSION" ]; then
-              errMsg="❌ Compiler version mismatch!%0AExpected: $COMPILER_VERSION%0AGot: $COMPUTED_COMPILER_VERSION"
+          # Check toolchain version
+          COMPILER_OUTPUT=$(compact compile --version)
+          COMPUTED_COMPILER_VERSION=$(echo "$COMPILER_OUTPUT" | grep -oP '\b0\.[0-9]+\.[0-9]+\b' | head -n 1)
```

Improves regex matching, currently fails to match 1.0.0 case. Side note, found this great online [tool](https://regexr.com/) that explains regexes character by character. Super useful!
```suggestion
          COMPUTED_COMPILER_VERSION=$(echo "$COMPILER_OUTPUT" | grep -oP '\b[0-9]+\.[0-9]+\.[0-9]+\b' | head -n 1)
```